harness/determined/experimental/client.py

```diff
@@ -225,14 +226,21 @@ def whoami() -> User:
     return _determined.whoami()
 
 
-@_require_singleton
+# DOES NOT REQUIRE SINGLETON (don't force a login in order to log out).
 def logout() -> None:
     """
     Get the :class:`~determined.experimental.client.User` representing the
     current user.
     """
-    assert _determined is not None
-    return _determined.logout()
+    if _determined is not None:
```

Comment:
This means if you lose your det singleton, you cannot log that session out, right?

Reply:
No, since the session is cached in the filesystem, you could either do:

- `client.Determined().logout()`, if it was the active session, or
- `client.Determined(master=..., user=...).logout()` if it was not the active session.

Either of those would pull the exact token out of token storage and then log it out.